File: C++/05. 1차원 배열/10810_공 넣기.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main()
{
    int N, M;
    cin >> N >> M;

    vector<int> basket(N);

    for (int i = 0; i < M; i++)
    {
        int start, end, number;
        cin >> start >> end >> number;

        fill(basket.begin() + start - 1, basket.begin() + end, number);
    }

    for (const auto &i : basket)
        cout << i << ' ';

    return 0;
}

File: Python/04. 1차원 배열/10810_공 넣기.py
N, M = map(int, input().split())

basket = [0 for _ in range(N)]

for i in range(M):
    start, end, number = map(int, input().split())
    basket[start - 1:end] = [number] * (end - start + 1)

for i in basket:
    print(i, end=" ")
